### PR TITLE
[WasmFS] Implement [f]statfs64 syscalls

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: system/lib/wasmfs/syscalls.cpp [hunk 1]
```diff
@@ -1389,6 +1390,44 @@ int __syscall_fcntl64(int fd, int cmd, ...) {
   }
 }
 
+int doStatFS(std::shared_ptr<File>& file, size_t size, struct statfs* buf) {
```
Comment (sbc100) [inline system/lib/wasmfs/syscalls.cpp:1393]: static?

**Review by sbc100 — APPROVED** [review]

**Review by tlively — APPROVED** [review]
Great, thanks! Should be setting fs_type as well?

File: system/lib/wasmfs/syscalls.cpp (hunk 1 above)
Comment (kripken) [reply system/lib/wasmfs/syscalls.cpp:1393]: Sounds good, will fix other cases without it too.

**Comment by kripken** [comment]
> Should be setting fs_type as well?

I'm not sure... the old FS didn't, and reading [the docs](https://man7.org/linux/man-pages/man2/statfs.2.html) I suspect there isn't an obvious value to return which is maybe why the old FS did nothing. What did you have in mind?

**Comment by tlively** [comment]
storing the backend_t in fs_type would be the most useful, but otherwise storing 0 would be better than leaving it undefined.

**Comment by kripken** [comment]
Good point, 0 is better than leaving it as whatever old value is there... fixed.

I'm not sure if `backend_t` makes sense or not there. It seems like we should define constants for the things, if we want them there. But let's leave that for later.

**Pull request merged**